server/mailet/integration-testing/src/test/java/org/apache/james/mailets/NetworkMatcherIntegrationTest.java

```diff
@@ -0,0 +1,324 @@
+    private static final int IMAP_PORT = 1143;
+    private static final String PASSWORD = "secret";
+
+    private static final String JAMES_APACHE_ORG = "james.org";
+    private static final String FROM = "fromuser@" + JAMES_APACHE_ORG;
+
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private TemporaryJamesServer jamesServer;
+    private ConditionFactory calmlyAwait;
+
+    @Before
+    public void setup() throws Exception {
+        Duration slowPacedPollInterval = Duration.FIVE_HUNDRED_MILLISECONDS;
+        calmlyAwait = Awaitility.with()
+            .pollInterval(slowPacedPollInterval)
+            .and().with()
+            .pollDelay(slowPacedPollInterval)
+            .await();
+    }
+
+    private TemporaryJamesServer createJamesServerWithRootProcessor(ProcessorConfiguration.Builder rootProcessor) throws Exception {
+        MailetContainer mailetContainer = MailetContainer.builder()
+            .postmaster("postmaster@" + JAMES_APACHE_ORG)
+            .threads(5)
+            .addProcessor(rootProcessor.build())
+            .addProcessor(CommonProcessors.error())
+            .addProcessor(deliverOnlyTransport())
+            .build();
+
+        TemporaryJamesServer temporaryJamesServer = new TemporaryJamesServer(temporaryFolder, mailetContainer);
+
+        DataProbe dataProbe = temporaryJamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(JAMES_APACHE_ORG);
+        dataProbe.addUser(FROM, PASSWORD);
+        return temporaryJamesServer;
+    }
+
+    private ProcessorConfiguration deliverOnlyTransport() {
+        return ProcessorConfiguration.builder()
+            .state("transport")
+            .enableJmx(true)
+            .addMailet(MailetConfiguration.builder()
+                .matcher(All.class)
+                .mailet(RemoveMimeHeader.class)
+                .addProperty("name", "bcc")
+                .build())
+            .addMailet(MailetConfiguration.builder()
+                .matcher(RecipientIsLocal.class)
+                .mailet(VacationMailet.class)
+                .build())
+            .addMailet(MailetConfiguration.builder()
+                .matcher(All.class)
+                .mailet(LocalDelivery.class)
+                .build())
+            .build();
+    }
+
+    private MailetConfiguration dropAllMailMailet() {
+        return MailetConfiguration.builder()
+            .matcher(All.class)
+            .mailet(Null.class)
+            .build();
+    }
+
+    @After
+    public void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void mailsFromAuthorizedNetworksShouldBeDeliveredWithRemoteAddrInNetwork() throws Exception {
+        jamesServer = createJamesServerWithRootProcessor(ProcessorConfiguration.builder()
+            .state("root")
+            .addMailet(MailetConfiguration.builder()
+                .matcher(RemoteAddrInNetwork.class)
+                .matcherCondition("127.0.0.1/32")
```

Comment:
I'm not sure it will work on every computer

Reply:
Good question. Better ideas?

Reply:
localhost is defined by 127.0.0.0/8, so maybe you can base only on this assumption.

Reply:
Ok I will